lib/fastlane/plugin/firebase_app_distribution/helper/firebase_app_distribution_auth_client.rb

```diff
@@ -48,20 +48,27 @@ def fetch_auth_token(google_service_path, firebase_cli_token, debug = false)
       private
 
       def refresh_token_from_firebase_tools
-        if ENV["XDG_CONFIG_HOME"].nil? || ENV["XDG_CONFIG_HOME"].empty?
-          config_path = File.expand_path(".config/configstore/firebase-tools.json", "~")
-        else
-          config_path = File.expand_path("configstore/firebase-tools.json", ENV["XDG_CONFIG_HOME"])
-        end
-
+        config_path = format_config_path
         if File.exist?(config_path)
-          begin
-            refresh_token = JSON.parse(File.read(config_path))['tokens']['refresh_token']
-            refresh_token unless refresh_token.nil? || refresh_token.empty?
-          # TODO: Catch parser errors, improve error handling here
-          # Returns nil when there is an empty "tokens" field in the firebase-tools json
-          rescue NoMethodError
+         begin
+          firebase_tools_tokens = JSON.parse(File.read(config_path))['tokens']
+          if (firebase_tools_tokens.nil?)
+           UI.important("Unable to find \'tokens\' field in file located at #{config_path} Ensure that the file has a tokens field and try again")
+           return
           end
+          refresh_token = firebase_tools_tokens['refresh_token']
```

Comment:
what if the 'refresh_token' is missing? I guess that should also produce a helpful error message. Or is that reported on a higher level if this function returns `nil`?

Reply:
If its nil thats fine we have a check on 63 that checks if its nil or empty. We just cant have tokens be nil or empty since we try to get the refresh token from it.